curp/src/server/cmd_worker/conflict_checked_mpmc.rs

```diff
@@ -404,8 +438,13 @@ impl<C: Command> Filter<C> {
                     let v = self.get_vertex_mut(vid);
                     match v.inner {
                         VertexInner::Cmd { ref mut as_st, .. } => {
-                            debug_assert!(matches!(*as_st, AsState::NotSynced));
-                            *as_st = AsState::AfterSyncReady(index);
+                            if let AsState::NotSynced(prepare) = as_st.clone() {
+                                *as_st = AsState::AfterSyncReady(index, prepare);
+                            } else {
+                                unreachable!(
+                                    "after sync state should be AsState::NotSynced but found {as_st:?}"
+                                );
+                            }
```

Comment:
Do you mean to modify it as:
```suggestion
                            if let AsState::NotSynced(prepare) = as_st {
                                *as_st = AsState::AfterSyncReady(index, prepare.take());
                            } else {
                                unreachable!(
                                    "after sync state should be AsState::NotSynced but found {as_st:?}"
                                );
                            }
```
Actually, it will against the `clippy::pattern_type_mismatch`:
```bash
error: type of pattern does not match the expression type
   --> curp/src/server/cmd_worker/conflict_checked_mpmc.rs:441:36
    |
441 | ...                   if let AsState::NotSynced(prepare) = as_st {
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: use `*` to dereference the match expression or explicitly match against a `&mut _` pattern and adjust the enclosed variable bindings
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#pattern_type_mismatch
note: the lint level is defined here
```

Reply:
Can't we fix this lint? if we can't, allow this clippy is better than clone as_st

Reply:
fix like this
```rust
if let AsState::NotSynced(ref mut prepare) = *as_st {
    *as_st = AsState::AfterSyncReady(index, prepare.take());
} else {
    unreachable!(
        "after sync state should be AsState::NotSynced but found {as_st:?}"
    );
}
```

Reply:
Done